### PR TITLE
Implements the "map_*_to_font" series of methods to Console::Root

### PR DESCRIPTION
**Pull request by obsoke**
As discussed in #244, Adds the following methods to `Root`:

* `map_ascii_code_to_font`
* `map_ascii_codes_to_font`
* `map_string_to_font`

No tests included in this PR (yet) as I'm unsure as to how to go about testing these methods . It seems like you need to use your eyes to see whether the correct font character was rendered.

Please let me know if anything is missing / needs fixing!

### CODE REVIEW

File: src/console.rs [hunk 1]
```diff
@@ -345,6 +345,52 @@ impl Root {
...
+                nb_codes,
+                font_char_x,
+                font_char_y
+            );
+        }
+    }
+
+    /// Maps ASCII codes from a string to consecutive characters in a bitmap font.
+    pub fn map_string_to_font(&self,
+                              s: &'static str,
```
Comment (Gustorn) [inline src/console.rs]: You don't need the `'static` lifetime here (and it actually limits some valid use-cases).

**Comment by Gustorn** [comment]
All of these should take `&mut self` instead of `&self`: if you look at the [original code](https://github.com/tomassedovic/tcod-rs/blob/master/tcod_sys/libtcod/src/sys_sdl_c.c#L215) there's some mutation going on which we should reflect in our API.

**Comment by tomassedovic** [comment]
Thanks! I can merge this as is, but there are two ways of improving it.

First, explain the function parameters in the documentation (the `///` comments turn into the docs). It's mostly self-explanatory here, but still.

Second, you're right there's no easy way to test this automatically, but we could write an example. That's good because it shows how to use the feature and makes sure we don't break the API by accident.

You can see this:

https://github.com/tomassedovic/tcod-rs/blob/master/examples/chars.rs

And run it as:

    cargo run --example chars

**Comment by obsoke** [comment]
Ah, an example is a good idea in lieu of a decent testing method. I will add that & expand the doc comments. Thanks to both of you for the feedback so far!

**Comment by obsoke** [comment]
I've added an explanation of function arguments, as well as a working example. I have to admit that the example is a bit contrived. Personally, I think showing off the use of a graphical file font (my original use case) would be neat but this would require adding a new font to the project. What do you folks think? Is it worth having a cooler example, or is this Good Enough? Also, please let me know if anything needs fixing up.

**Comment by tomassedovic** [comment]
Thanks!

I agree that adding a custom font would be much better (and is what these functions are designed for) and so long as there are no licensing issues with the font, I'd prefer that. Something like creative commons or public domain should be fine.

But if you want to merge this now (it looks good to me), we can do that and add a better example in a later pull request. Whichever you prefer.

**Comment by obsoke** [comment]
angband16x16.bmp is in public domain, [according to its creator](http://angband.oook.cz/forum/showpost.php?p=316&postcount=16).

**Comment by christopherdumas** [comment]
In my issue #247, I have a tileset that looks pretty nice (in my opinion). It was drawn by a friend of mine. You could use that, although it is possibly broken or incorrectly formatted (see the issue).

**Comment by christopherdumas** [comment]
I would really like this merged soon, because right now I'm writing a game that is using the `map_*_to_font` functions in unsafe mode and it would be nice to not have to use unsafe.

**Comment by obsoke** [comment]
Is there anything else you folks need in this PR? I've updated it to have a less contrived example & added a public domain graphical tileset.

**Comment by tomassedovic** [comment]
I'm really sorry it took me so long. This looks great, merging it now.

**Pull request merged**